pkg/ipamd/ipamd.go

```diff
@@ -501,14 +507,30 @@ func (c *IPAMContext) getLocalPodsWithRetry() ([]*k8sapi.K8SPodInfo, error) {
 // StartNodeIPPoolManager monitors the IP pool, add or del them when it is required.
 func (c *IPAMContext) StartNodeIPPoolManager() {
 	sleepDuration := ipPoolMonitorInterval / 2
+	// If no whitelist is defined, or whitelist contains cached owner id, ENI/IPs need to be tracked.
+	// If owner id is not set, or not in the whitelist, don't do ENI tracking.
+	shouldWhitelistTheAccount :=
+		c.accountWhitelist == nil || contains(c.accountWhitelist, c.awsClient.GetAccountId())
+
 	for {
 		time.Sleep(sleepDuration)
-		c.updateIPPoolIfRequired()
+		if shouldWhitelistTheAccount {
+			c.updateIPPoolIfRequired()
+		}
```

Comment:
I think this is not the right place to check this. Maybe we need a more clear specification of the ask here. All node local updates still need to happen, but calls to the EC2 API should be avoided, so the checks need to be more granular.